modules/kargoAnalyticsAdapter.js

```diff
@@ -1,14 +1,98 @@
+import { logError } from '../src/utils.js';
+import { ajax } from '../src/ajax.js';
 import adapter from '../src/AnalyticsAdapter.js';
 import adapterManager from '../src/adapterManager.js';
+import CONSTANTS from '../src/constants.json';
 
-var kargoAdapter = adapter({
-  analyticsType: 'endpoint',
-  url: 'https://krk.kargo.com/api/v1/event/prebid'
-});
+const EVENT_URL = 'https://krk.kargo.com/api/v1/event';
+const KARGO_BIDDER_CODE = 'kargo';
+
+const analyticsType = 'endpoint';
+
+let _initOptions = {};
+
+let _logBidResponseData = {
+  auctionId: '',
+  auctionTimeout: 0,
+  responseTime: 0,
+};
+
+let _bidResponseDataSent = false;
+
+var kargoAnalyticsAdapter = Object.assign(
+  adapter({ analyticsType }), {
+    track({ eventType, args }) {
+      switch (eventType) {
+        case CONSTANTS.EVENTS.AUCTION_INIT: {
+          _logBidResponseData.auctionTimeout = args.timeout;
+          break;
+        }
+        case CONSTANTS.EVENTS.BID_RESPONSE: {
+          handleBidResponseData(args);
+          break;
+        }
+        case CONSTANTS.EVENTS.AUCTION_END: {
+          // Reset bool for whether or not to send bid response data
+          _bidResponseDataSent = false;
+          break;
+        }
+      }
+    }
+  }
+);
+
+function handleBidResponseData (bidResponse) {
+  if (bidResponse.bidder !== KARGO_BIDDER_CODE || _bidResponseDataSent) {
+    return
+  }
+
+  _logBidResponseData.auctionId = bidResponse.auctionId;
+  _logBidResponseData.responseTime = bidResponse.responseTimestamp - bidResponse.requestTimestamp;
```

Comment:
I think prebid core sets a param on this object called `timeToRespond` which is essentially this.

Also, you want to only track when your exchange does a VALID bid response? Not when it no-bids or errors?